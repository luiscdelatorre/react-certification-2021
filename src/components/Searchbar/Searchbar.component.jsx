import React, { useState } from 'react';
import { IoSearchOutline } from 'react-icons/io5';
import { Search, SearchInput } from './Searchbar.styles';

const Searchbar = () => {
  const [isSearchActive, setIsSearchActive] = useState(false);
  const [searchValue, setSearchValue] = useState('');

  const toggleSearch = () => {
    setIsSearchActive(searchValue ? true : !isSearchActive);
  };

  const search = (event) => {
    event.preventDefault();
    console.log(searchValue);
  };

  const handleChange = (event) => {
    setSearchValue(event.target.value);
  };

  const handleBlur = () => {
    setIsSearchActive(searchValue);
  };

  return (
    <Search onClick={toggleSearch} data-testid="searchbar">
      <IoSearchOutline />
      <form onSubmit={search} data-testid="form-search">
        <SearchInput
          value={searchValue}
          onChange={handleChange}
          onBlur={handleBlur}
          isSearchActive={isSearchActive}
          type="search"
          placeholder="Search"
          data-testid="input-search"
        />
      </form>
    </Search>
  );
};

export default Searchbar;
