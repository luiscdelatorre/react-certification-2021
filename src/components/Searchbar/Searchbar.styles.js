import styled from 'styled-components';
import device from '../../config/device';

const Search = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 2rem;
  position: relative;
  svg {
    color: ${(props) => props.theme.textSecondary};
    z-index: 1;
    position: absolute;
    left: 1.5rem;
    @media ${device.laptop} {
    }
  }
`;

const SearchInput = styled.input`
  position: relative;
  color: ${(props) => props.theme.textSecondary};
  padding-left: 5rem;

  @media ${device.onlySm} {
    border-color: ${(props) =>
      props.isSearchActive ? props.theme.borderColor : 'transparent'};
    padding-right: ${(props) => (props.isSearchActive ? '1.5rem' : '0')};
    padding-left: ${(props) => (props.isSearchActive ? '5rem' : '3rem')};
    width: ${(props) => (props.isSearchActive ? '100%' : '0')};
    &:focus {
      padding-right: 1.5rem;
      padding-left: 5rem;
      width: 100%;
    }
  }
  @media ${device.laptop} {
    width: 300px;
  }
`;

export { Search, SearchInput };
