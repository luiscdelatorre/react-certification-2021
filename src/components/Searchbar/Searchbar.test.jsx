import { fireEvent, getByTestId, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import Searchbar from './Searchbar.component';

describe('Searchbar component', () => {
  it('should focus input clicking on searchbar', () => {
    const { container } = render(<Searchbar />);
    const searchbar = getByTestId(container, 'searchbar');
    const input = getByTestId(container, 'input-search');
    const initialClassName = input.className;
    fireEvent.click(searchbar);
    expect(initialClassName).not.toBe(input.className);
  });

  it('should trigger input events: focus, change and blur', () => {
    const { container } = render(<Searchbar />);
    const searchbar = getByTestId(container, 'searchbar');
    const input = getByTestId(container, 'input-search');
    const initialClassName = input.className;

    fireEvent.click(searchbar);
    userEvent.type(input, 'Foo');
    expect(input.value).toBe('Foo');

    fireEvent.blur(input);
    fireEvent.click(searchbar);
    expect(initialClassName).not.toBe(input.className);
  });

  it('should trigger search event', () => {
    const { container } = render(<Searchbar />);
    const searchbar = getByTestId(container, 'searchbar');
    const input = getByTestId(container, 'input-search');
    const form = getByTestId(container, 'form-search');

    jest.spyOn(global.console, 'log');
    fireEvent.click(searchbar);
    userEvent.type(input, 'Foo');
    fireEvent.submit(form);

    expect(console.log).toHaveBeenCalled();
  });
});
