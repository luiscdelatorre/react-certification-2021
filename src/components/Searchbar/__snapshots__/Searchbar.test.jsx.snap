// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar component Validates VideoCard snapshot 1`] = `
<div
  className="sc-bdfBwQ gmamkS"
  data-testid="searchbar"
  onClick={[Function]}
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M221.09 64a157.09 157.09 0 10157.09 157.09A157.1 157.1 0 00221.09 64z"
      fill="none"
      strokeMiterlimit="10"
      strokeWidth="32"
    />
    <path
      d="M338.29 338.29L448 448"
      fill="none"
      strokeLinecap="round"
      strokeMiterlimit="10"
      strokeWidth="32"
    />
  </svg>
  <form
    data-testid="form-search"
    onSubmit={[Function]}
  >
    <input
      className="sc-gsTCUz eSCwiF"
      data-testid="input-search"
      onBlur={[Function]}
      placeholder="Search"
      type="search"
    />
  </form>
</div>
`;
